packages/api/src/controllers/access-control.ts

```diff
@@ -8,4 +14,66 @@ const app = Router();
 accessControl.use("/signing-key", signingKeyApp);
 app.use("/access-control", accessControl);
 
+accessControl.post(
+  "/gate",
+  authorizer({ anyAdmin: true }),
+  validatePost("access-control-gate-payload"),
+  async (req, res) => {
+    const playbackId = req.body.stream.replace("video+", "");
+    const content =
+      (await db.stream.getByPlaybackId(playbackId)) ||
+      (await db.asset.getByPlaybackId(playbackId));
+
+    if (!content) {
+      throw new ForbiddenError("Content not found");
+    }
+
+    if (content.deleted) {
+      throw new ForbiddenError("Stream is deleted");
+    }
+
+    const playbackPolicy = content.playbackPolicy;
+
+    if (playbackPolicy) {
+      if (playbackPolicy.type === "public") {
+        res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
+        res.status(204);
+        return res.end();
+      }
+
+      if (playbackPolicy.type === "signed") {
+        if (!req.body.pub) {
+          throw new ForbiddenError("Stream is gated and requires a public key");
+        }
+
+        const query = [];
+        query.push(sql`signing_key.data->>'publicKey' = ${req.body.pub}`);
+        const [signingKeyOutput] = await db.signingKey.find(query, {
+          limit: 1,
+        });
+
+        if (signingKeyOutput.length == 0) {
+          throw new ForbiddenError(
```

Comment:
This should be a 404 as well wdyt?

Reply:
Even for missing signing keys? I would say a `403` with `The content is gated but the corresponding Signing Key was not found` would be more precise, as I would expect `404` to be for a missing content

Reply:
Makes sense!